src/wordlift/post-excerpt/class-post-excerpt-rest-controller.php

```diff
@@ -0,0 +1,223 @@
+	 */
+	const WORDLIFT_POST_EXCERPT_URL = 'https://api.wordlift.io/summarize';
+
+	/**
+	 * Wordlift returns excerpt in response using this key..
+	 */
+	const WORDLIFT_POST_EXCERPT_RESPONSE_KEY = 'summary';
+
+	public static function register_routes() {
+		add_action( 'rest_api_init', 'Wordlift\Post_Excerpt\Post_Excerpt_Rest_Controller::register_route_callback' );
+	}
+
+	/**
+	 * Saves the excerpt in the post meta.
+	 *
+	 * @param $post_id int Post id which the post excerpt belongs to
+	 * @param $post_excerpt string Post excerpt returned by the server
+	 * @param $post_body string Total text content of the post body.
+	 *
+	 * @return void
+	 */
+	public static function save_post_excerpt_in_meta( $post_id, $post_excerpt, $post_body ) {
+		// hash the post body and save it.
+		$data = array(
+			'post_body_hash' => md5( $post_body ),
+			'post_excerpt'   => $post_excerpt
+		);
+		update_post_meta( $post_id, self::POST_EXCERPT_META_KEY, $data );
+	}
+
+	/**
+	 * Sends the remote request to the wordlift API and saves the response in meta for
+	 * future use.
+	 *
+	 * @param $post_id int Post id which the post excerpt belongs to
+	 * @param $post_body string Total text content of the post body.
+	 *
+	 * @return array|bool
+	 */
+	public static function get_post_excerpt_from_remote_server( $post_id, $post_body ) {
+		// The configuration is constant for now, it might be changing in future.
+		$configuration         = array(
+			'ratio'      => 0.0005,
+			'min_length' => 60
+		);
+		$configuration_service = \Wordlift_Configuration_Service::get_instance();
+		// Construct the url with the configuration
+		$url      = add_query_arg( $configuration, self::WORDLIFT_POST_EXCERPT_URL );
+
+		$response = wp_remote_post( $url, array(
+			'timeout'    => 60,
+			'user-agent' => User_Agent::get_user_agent(),
+			'headers'    => array(
+				'Content-Type'  => 'text/plain',
+				'Authorization' => "Key {$configuration_service->get_key()}",
+			),
+			'body'       => $post_body,
+			'data_format' => 'body',
+		) );
+		return self::save_response_to_meta_on_success( $post_id, $post_body, $response );
+	}
+
+	/**
+	 * Determines whether we need to get the excerpt from wordlift api,
+	 * or just use the one we already obtained by generating the hash and comparing it
+	 * with the previous one.
+	 *
+	 * @param $request WP_REST_Request $request {@link WP_REST_Request instance}.
+	 *
+	 * @return array Post excerpt data.
+	 */
+	public static function get_post_excerpt( $request ) {
+		$data            = $request->get_params();
+		$post_id         = $data['post_id'];
+		$post_body       = $data['post_body'];
+		$current_hash    = md5( $post_body );
+		$server_response = self::get_post_excerpt_conditionally( $post_id, $post_body, $current_hash );
+		if ( $server_response === NULL || ! array_key_exists( 'post_excerpt', $server_response ) ) {
+			return array(
+				'status' => 'error',
+				'message' => __('Unable to contact wordlift server', 'wordlift')
```

Comment:
@naveen17797 `WordLift` not `wordlift`, i.e. change to  `__('Unable to contact WordLift API', 'wordlift')`